docs/Behaviour - Client Side.md

```diff
@@ -13,14 +13,14 @@ Constraint Sets of Active Constraints SHOULD represent a consensus among Receive
 Constraint Sets of Receiver Capabilities with `urn:x-nmos:cap:meta:enabled` set to false MUST be ignored completely while making the processing.
 
 
-## Dynamic format changes on Sender
+## Dynamic Update of Resources
 
-### Changes which break Active Constraints
+The Controller MUST be capable of using Stream Compatibility API in conjunction with IS-04 Query API (if available) or IS-04 Node API to discover and dynamically report the state of any relevant resources, including Inputs, Outputs, Senders and Receivers.
 
-NMOS Controller MUST track Sender's `subscription`'s `active` property during and after the IS-05 connection. If it became `false` unexpectedly, NMOS Controller SHOULD `GET /status` to figure out the reason of breaking the connection. If the State of the Sender is `active_constraints_violation`, then NMOS Controller SHOULD either disconnect all the Receivers or `PUT /constraints/active` once again and re-enable Sender via IS-05.
```

Comment:
You removed the "tracking sender subscription active property" aspect ... We must favor monitoring through the registry query/subscription mechanism first ... I don't think it would be a good idea for Controller to poll the status/state. In all aspects of Controller monitoring we must make sure that it is being performed not by polling the device but the registry.

Reply:
If I correctly understood Gareth in #116, it's implied that Controller tracks resources' properties, we need to dictate what Controller should do with their values. Moreover, `active` property update is a consequence of changing the State of a Sender to `active_constraints_violation` so maybe this is what Controller should track instead.

[The current text](https://specs.amwa.tv/is-11/branches/v1.0-dev/docs/Interoperability.html#version-increments) says:
>When State of Sender or Receiver is changed, or when the set of Inputs/Outputs associated with the Sender/Receiver is changed, then the version attribute of the relevant IS-04 Sender or Receiver MUST be incremented.

Do I correctly understand, that it means that Controller doesn't have to periodically poll `/status` and can it do that only after gets `version` update from Query API?

>In all aspects of Controller monitoring we must make sure that it is being performed not by polling the device but the registry.

I tried to express it with `The Controller MUST be capable of using Stream Compatibility API in conjunction with IS-04 Query API (if available) or IS-04 Node API`. [IS-04 RECOMMENDs](https://specs.amwa.tv/is-04/releases/v1.3.2/docs/Controllers.html#websocket-subscriptions) to use WebSocket subscriptions. Inserted this paragraph as a suggestion.

Reply:
"Do I correctly understand, that it means that Controller doesn't have to periodically poll /status and can it do that only after gets version update from Query API?"

What the text says is a requirement for the Node.

"In all aspects of Controller monitoring we must make sure that it is being performed not by polling the device but the registry."

Where is this sentence coming from?

At the IS-11 level we must make sure people will not mis-interpret our specification because they haven't read the full IS-04 and/or Controller specifications and emphasis when possible that because an IS-11 Node have a requirement to update the Sender's version is various cases, the Controller SHOULD monitor from the Registry when available.